File: 2125-number-of-laser-beams-in-a-bank/2125-number-of-laser-beams-in-a-bank.cpp
class Solution {
public:
    int numberOfBeams(vector<string>& bank) {
        int n=bank.size(); int m=bank[0].size(); vector<int>nums;  int cnt=0; int ans=0; 
        for(int i=0;i<n;i++)
        {
            for(int j=0;j<m;j++)
            {
                if(bank[i][j]=='1')
                {
                   cnt++;
                }
            }
            if(cnt>0) nums.push_back(cnt);
            cnt=0;
        } 
        for(int i=1;i<nums.size();i++)
        {
            ans+=nums[i-1]*nums[i];
        }
        return ans;
    }
};

File: 2125-number-of-laser-beams-in-a-bank/2125-number-of-laser-beams-in-a-bank.java
class Solution {
    public int numberOfBeams(String[] bank) {
        String s=bank[0]; int cnt=0;
        for(int i=0;i<s.length();i++)
        {
            if(s.charAt(i)=='1') cnt++;
        }
        int ans=0;
        for(int i=1;i<bank.length;i++)
        {
            String str=bank[i]; 
            int cnt1=0;
            for(int j=0;j<str.length();j++)
            {
                if(str.charAt(j)=='1') cnt1++;
            }
            if(cnt1!=0) 
            {
            ans+=cnt*cnt1; 
            cnt=cnt1; 
            }
        }
        return ans;
    }
}